### ISSUE TITLE
ArrayWidget shouldn't allow creatable

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#2986]
If the ArrayWidget is used for "creatable terms", then there's a mixup between tokens, labels and values, which makes the creatable feature useless (basically, you can create the terms, they look fine in the edit view, but in the object they are saved with autogenerated tokens, which are meaningless as a value).

@giuliaghisini Am I correct, or am I missing something?

**Comment by giuliaghisini** [comment]
i agree, but we had this case https://github.com/collective/volto-form-block/blob/1cc2ad6558585eba3dcbca53475199a17ab97d94/src/components/Sidebar.jsx#L539
where we want enable user to set a simple list of terms. And ArrayWidget with creatable is right for us.. or do you have any other suggestion?

**Comment by tiberiuichim** [comment]
@giuliaghisini It's better to use the TokenWidget. I think the ArrayWidget should follow the same conventions as the TokenWidget (use the label of the created element as the "token"), and when that happens, what will be the difference to the TokenWidget? I think they're the same widget.

**Comment by giuliaghisini** [comment]
I think there is some general confusion with these widgets, maybe we should take stock of the situation and document the differences of these widgets somewhere (here https://docs.voltocms.com/ for example). In this specific case I don't even have a vocabulary associated with the widget, it's a pure and simple list of strings. I'll take a look to the TokenWidget

**Comment by tiberiuichim** [comment]
@giuliaghisini I've started to do that here https://docs.voltocms.com/recipes/widget/ and I've added a ton of stories for the widgets. https://docs.voltocms.com/storybook/?path=/story/widgets-array--creatable&globals=measureEnabled:false

Actually now looking at the array widget creatable story, it looks like the array widget does the right thing? Meh... I need to check what happened there, maybe it's backend widget configuration, frontend older version, I don't know.

**Comment by giuliaghisini** [comment]
exactly! i tested the token widget, it do the same of the array widget, but it needs a vocabulary. Array widget, on the other hand, doesn't need a vocabulary. And that's exactly my case: having a creatable array without an attached vocabulary. 
Maybe, what we could do, is to unify array widget and token widget in a single widget..
